src/libopensc/card-piv.c

```diff
@@ -2965,7 +3213,47 @@ static int piv_pin_cmd(sc_card_t *card, struct sc_pin_cmd_data *data,
 			}
 		}
 	}
-	return iso_drv->ops->pin_cmd(card, data, tries_left);
+
+	priv->pin_cmd_verify_sw1 = 0x00U;
+
+	priv->pin_cmd_verify = 1; /* tell piv_check_sw its a verify to save sw1, sw2 */
+	r = iso_drv->ops->pin_cmd(card, data, tries_left);
+	priv->pin_cmd_verify = 0;
+
+	/* if access to applet is know to be reset by other driver  we select_aid and try again */
+	if ( priv->card_issues & CI_OTHER_AID_LOSE_STATE && priv->pin_cmd_verify_sw1 == 0x6DU) {
+		sc_debug(card->ctx, SC_LOG_DEBUG_NORMAL, "AID may be lost doing piv_find_aid and retry pin_cmd");
```

Comment:
`priv->pin_cmd_verify_sw1 == 0x6DU` will always be false because you're setting `priv->pin_cmd_verify_sw1 = 0x00U;` a couple of lines before and `r = iso_drv->ops->pin_cmd(card, data, tries_left)` won't change this value